src/js/components/App.js

```diff
@@ -69,9 +69,10 @@ class App extends Component {
       head(projects)
 
     const task =
-      findTask(this.changeset.task_id || service?.taskId || lastTaskId)(
-        project
-      ) || head(project?.tasks)
+      findTask(this.changeset.task_id || service?.taskId || lastTaskId)(project) ||
+      head(project?.tasks)
+
+    const billable = /\(.+\)/.test(this.changeset.hours) === true ? false : !!task?.billable
```

Comment:
Not really related to this PR, but in general: there's quite a few objects where you use `?` to access properties and it's not clear to me why they would be nullable? Can you elaborate?

I would further extract all these `consts` to methods, what do you think?

```js
get isBillable() {
  const isHoursInBrackets = /\(.+\)/.test(this.changeset.hours)
  return isHoursInBrackets ? false : task.billable
}
```

What do you think?

Reply:
Some of the `const`s are needed later, e.g. `project` is needed for setting `task`. Billable could be extracted, but honestly I do not see the benefit of that, as it is not reused anywhere else. But of course, I agree with you that the code can be refactored and improved. The idea was that I present you the code and then it can be improved in a review. @agileapplications Maybe we can reserve some time in the next Thursday-meeting?

Regarding null-properties:
- when no service matches, the service is always undefined
- when there are no projects, project is undefined.
- when there is no project, or no tasks defined for a project, task is undefined.

Reply:
How can there be no projects? If you don't have any projects, you cannot track time, right? So you would have some kind of placeholder view?

If the service is undefined, I would introduce some kind of `NullService` or `DefaultService`. Maybe even an empty object is enough.

> `project` is needed for the calculation of `task`, ...

Even if they have dependencies on each other, they could still all call the methods and since they would be `@computed` I assume, no?

This is nothing huge and I don't to want to complicate things here, but I would suggest iterative improvements to make things nicer as we go. What do you think @agileapplications?

Reply:
An empty object or a NullService would complicate things elsewhere, e.g. the bubble is unmounted if the service is truthy and unmounted otherwise. An empty object or a NullService would always be truthy.

There is no placeholder view when there a no projects. The selects for projects/tasks are empty and the form cannot be submitted without a valid project and task.

Yes, you are right, the methods could call each other. But as I stated above, I think it's better not to start refactoring now. Let's refactor after a full review of the code and architecture. We'll have a proper view of the whole app then and it might be that the method `changesetWithDefaults` will need to be replaced with a different concept. All the refactoring we do now would be for nothing.